src/classes/RD2_OpportunityService_TEST.cls

```diff
@@ -199,6 +199,33 @@ private with sharing class RD2_OpportunityService_TEST {
         testAndAssertOppService(rd, expectedOppName);
     }
 
+    /***
+    * @description Verifies the installment opportunities are created with the record types set in the NPSP setting
+    */
+    @isTest
+    private static void oppRecordTypeShouldMatchWithSetting() {
```

Comment:
Let's additionally create a unit test to verify behaviour when record type Id is not valid.